core/spot_price_test.go

```diff
@@ -124,3 +136,48 @@ func Test_spotPrices_average(t *testing.T) {
 		})
 	}
 }
+
+func Test_fetch(t *testing.T) {
+	tests := []struct {
+		name             string
+		config           *spotPrices
+		product          string
+		duration         time.Duration
+		availabilityZone *string
+		instanceTypes    []*string
+		err              error
+	}{
+		{
+			name: "error",
+			config: &spotPrices{
+				data: []*ec2.SpotPrice{},
+				conn: connections{
+					ec2: mock{
+						resp: &ec2.DescribeSpotPriceHistoryOutput{},
+						er:   errors.New("Error!"),
+					},
+				},
+			},
+			err: errors.New("Error!"),
+		},
+		{
+			name: "ok",
+			config: &spotPrices{
+				data: []*ec2.SpotPrice{},
+				conn: connections{
+					ec2: mock{resp: &ec2.DescribeSpotPriceHistoryOutput{}},
+				},
+			},
+			err: errors.New(""),
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			err := tc.config.fetch(tc.product, tc.duration, tc.availabilityZone, tc.instanceTypes)
+			if err != nil && err.Error() != tc.err.Error() {
+				t.Errorf("error expected: %s, actual: %s", tc.err.Error(), err.Error())
+			}
+		})
+	}
+}
```

Comment:
As the purpose of the method is to fetch price, I wonder if it wouldn't make sense to actually add fake price data, that could be checked as well, in bose err/nil cases. WDYT?